src/universal/modules/welcome/components/WelcomeWizardForms/Step3InviteTeam.js

```diff
@@ -28,9 +28,37 @@ const emailInviteFail = emailsNotDelivered => ({
   level: 'error'
 });
 
-const Step3InviteTeam = props => {
-  const {dispatch, handleSubmit, invitees, inviteesRaw, submitting, teamName, welcome: {teamId}} = props;
-  const onAddInviteesButtonClick = event => {
+
+@reduxForm({
+  form: 'welcomeWizard',
+  destroyOnUnmount: false
+  // TODO: add sync + mailgun async validations
+})
+export default class Step3InviteTeam extends Component {
+  static propTypes = {
+    dispatch: PropTypes.func,
+    handleSubmit: PropTypes.func,
+    invitees: PropTypes.array,
+    inviteesRaw: PropTypes.string,
+    onSubmit: PropTypes.func,
+    submitting: PropTypes.bool,
+    teamName: PropTypes.string,
+    welcome: PropTypes.shape({
+      teamId: PropTypes.string,
+    })
+  };
+
+  shouldComponentUpdate(nextProps) {
+    const {teamName, submitting} = nextProps;
+    if (submitting & !teamName) {
+      // do not update if we're submitting
+      return false;
```

Comment:
Love the attention to detail! I think this is something that redux-form should handle. I'll write an issue.